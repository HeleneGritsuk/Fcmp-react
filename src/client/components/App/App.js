import { Link } from 'react-router';
import LinkContainer from 'react-router-bootstrap/lib/LinkContainer';
import React from 'react';
import Grid  from 'react-bootstrap/lib/Grid';
import Nav from 'react-bootstrap/lib/Nav';
import Navbar from 'react-bootstrap/lib/Navbar';
import NavItem  from 'react-bootstrap/lib/NavItem';

// import './bootstrap.css';

class App extends React.Component {
  constructor(props, context) {
    super(props, context);
  }
  render() {
    return (
      <div>
        <Navbar>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to='/'>Main</Link>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
          <Navbar.Collapse>
            <Nav navbar>
              <LinkContainer to='/login'>
                <NavItem>Log in</NavItem>
              </LinkContainer>
              <LinkContainer to='/register'>
                <NavItem>Register</NavItem>
              </LinkContainer>
              <LinkContainer to='/posts'>
                <NavItem>Posts</NavItem>
              </LinkContainer>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
        <Grid>
          {this.props.children}
        </Grid>
      </div>
    );
  }
}

export default App;
