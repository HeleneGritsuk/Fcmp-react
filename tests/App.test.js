import React from 'react';
import App from '../src/client/components/App';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('should render App correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
