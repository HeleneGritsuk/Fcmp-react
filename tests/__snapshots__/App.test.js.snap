// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App correctly 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          onClick={[Function]}
          style={Object {}}
        >
          Main
        </a>
        <button
          className="navbar-toggle collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <ul
          className="nav navbar-nav"
          role={null}
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Log in
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Register
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              action="push"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Posts
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  />
</div>
`;
